### ISSUE TITLE
How to change the language in betterbird?

### ISSUE THREAD

**Issue by scruel** [Betterbird/thunderbird-patches#40]
I'd like to change the language in betterbird itself, and no need to download many bin files with different languages, should it be possible in the future? And how to have a Simplified Chinese language in betterbird? I haven't seen it in your website download page, but exists here in your patch folder.
Thanks!

**Comment by Betterbird** [comment]
We're currently not supporting language packs to discourage the use of incompatible TB language packs which totally break BB so it doesn't even start. It's hard to recover from this issue, it involved to manually delete the language pack from the profile.

The Chinese localisation is incomplete:
https://github.com/Betterbird/thunderbird-patches/blob/main/scripts/zh-CN.cmd
so we don't ship it. It was originally created here (https://github.com/Betterbird/thunderbird-patches/pull/14) but the author hasn't localised anything after that. There also hasn't been any request for zh-CN so far.

**Comment by scruel** [comment]
> We're currently not supporting language packs to discourage the use of incompatible TB language packs which totally break BB so it doesn't even start. It's hard to recover from this issue, it involved to manually delete the language pack from the profile.
> 
> The Chinese localisation is incomplete: https://github.com/Betterbird/thunderbird-patches/blob/main/scripts/zh-CN.cmd so we don't ship it. It was originally created here (#14) but the author hasn't localised anything after that. There also hasn't been any request for zh-CN so far.

How can I know all the original strings in English? And can you show me how to do the localization job?

**Comment by Betterbird** [comment]
OK, Chinese versions here now: https://www.betterbird.eu/downloads/ - Let us know if something isn't right.

**Comment by scruel** [comment]
> OK, Chinese versions here now: https://www.betterbird.eu/downloads/ - Let us know if something isn't right.

Great work, I will check it soon.

**Comment by WillZegal** [comment]
Hi. I hope it's the good place for my question.
I've just installed Betterbird on my Mac. It's a new installation : I didn't had Thunderbird on this machine (I've not used TB for years).
It looks nice. But I can't change the language. I've downloaded the French Language package, but when I go to setting / general / language / Set Alternatives... "select a language to add" is greyed.

I didn't find the way anywhere else.

**Comment by Betterbird** [comment]
The language pack needs to be installed like an add-on:
![image](https://github.com/Betterbird/thunderbird-patches/assets/86832723/b1ed413b-8255-40cc-a308-29fd57f0fa76)

Then add the language in the settings:
![image](https://github.com/Betterbird/thunderbird-patches/assets/86832723/28601adc-c43e-450d-a07f-d3548185647b)

**Comment by WillZegal** [comment]
Thanks a lot, but I have no "language" option in add-ons
![Capture d’écran 2024-03-08 à 19 01 44](https://github.com/Betterbird/thunderbird-patches/assets/47359901/dc0150ed-dfa9-4a01-a6ce-168c152bd272)

**Comment by Betterbird** [comment]
That's because you haven't installed any language pack yet. Just drag the XPI onto the the Add-on Manager or use this menu:
![image](https://github.com/Betterbird/thunderbird-patches/assets/86832723/acb0181d-878e-4113-9e81-07791c515bf6)

**Comment by WillZegal** [comment]
Great. Problem solved. Thanks a lot for your kind help.

**Comment by Lister0fSmeg** [comment]
I am having problems with this myself. I am trying the Betterbird portable v115.17.0.bb35 with en-GB-115.11.0-bb28.xpi which is the latest that I can find.
I have tried dragging and dropping and also selecting install add add on from file in the extension display window. Neither of those do anything, they don't show any installation or show any additional language under language selection about a third the way down on the General tab. The Set Alternatives window accessed besides the English(US) under languages only shows English(US) and everything else is greyed out and nothing can be selected because the British English language doesn't seem to of been installed.

What is the purpose of this language pack if it does not work? Does it only work on the installed version of Betterbird?

**Comment by Betterbird** [comment]
Maybe you want to tone down your language a little. The language pack does work, as you can see here in a combination of 115.17.0-bb35 portable with en-GB-115.11.0-bb28.xpi. We used "Install Add-on From File...", but dropping should also work:

![image](https://github.com/user-attachments/assets/f260ab67-b1a5-44f0-8e44-05f20ac6522c)

![image](https://github.com/user-attachments/assets/71b9ba84-9b31-4ec2-9801-1e2093280eca)

![image](https://github.com/user-attachments/assets/d197f1db-de8d-4599-bfde-f4d0de3e6a64)

So what are you doing wrong or what is wrong on your system? Any clues in the Error Console as to why "Install Add-on From File..." doesn't work?

**Comment by Lister0fSmeg** [comment]
Thank you for helping out. I have now managed to get it installed today. I had not tried it again after I first failed to get it to install.

I think the problem was because of a very long delay when installing it, I run it in a sandbox which seems to affect it. There was a very long delay between adding the language pack and the tiny window asking to confirm the install. I must of missed that when I previously attempted to install it. I am guessing that I probably never confirmed the install because of the very long delay.

I am using an older Thunderbird sandbox folder which I copied and renamed to keep the original Thunderbird folder untouched. I am not sure why it is running so slow when using Sandboxie.

I did open the error console but didn't see anything relevant TBH. It was only when I saw the confirm install dialogue that it clicked why it hadn't worked previously, because I hadn't actually finished installing it. Something like 45 seconds is a long time to wait for it. Problem solved. I will have to look into why the sandbox is slowing it down so much.
